tox_binary_io/src/crypto.rs

```diff
@@ -0,0 +1,52 @@
+use std::convert::TryInto;
+
+use nom::{map, map_opt, named, take};
+
+use crypto_box::{PublicKey, SecretKey, KEY_SIZE};
+
+use super::FromBytes;
+
+
+impl FromBytes for PublicKey {
+    // TODO: use map_from with new nom version
+    named!(from_bytes<PublicKey>, map!(map_opt!(take!(KEY_SIZE), |pk: &[u8]| pk.try_into().ok()), |pk: [u8; KEY_SIZE]| pk.into()));
+}
+
+/* TODO
+Use the following implementation when https://github.com/TokTok/c-toxcore/issues/1169 is fixed.
```

Comment:
It is fixed

Reply:
Fixed, but I'd leave it for another MR, because firstly we need to check how many nodes from tox network migrated to a new version.